core/common/src/main/java/alluxio/wire/ConfigInfo.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.wire;
+
+import com.google.common.base.Objects;
+
+import java.io.Serializable;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * The Alluxio runtime configuration information.
+ */
+@NotThreadSafe
+public final class ConfigInfo implements Serializable {
+  private static final long serialVersionUID = 1025798648274314086L;
+
+  private String mName;
+  private String mValue;
+  private String mSource;
+
+  /**
+   * Creates a new instance of {@link ConfigInfo}.
+   */
+  public ConfigInfo() {}
+
+  /**
+   * Creates a new instance of {@link ConfigInfo} from a thrift representation.
+   *
+   * @param configInfo the thrift representation of Alluxio configuration information
+   */
+  protected ConfigInfo(alluxio.thrift.ConfigInfo configInfo) {
```

Comment:
`protected` is a little weird here since it usually means that we want subclasses to be able to use it, but `ConfigInfo` doesn't have any subclasses. It's cleaner to put the `toThrift` and `fromThrift` methods both in this class and keep the visibility `private`

Reply:
Changed to private and add toThrift and fromThrift inside ConfigProperty. Thanks